### PR TITLE
Add document describing Policy Testing concept

### PR DESCRIPTION
**Pull request by Springstone**
## Overview/Summary

Add document describing Policy Testing concept

## As part of this Pull Request I have

- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [ ] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Review by paulgrimley — APPROVED** [review]
Great work @Springstone LGTM from a high level read and very valuable addition based on requests to understand this further from the community

File: docs/wiki/ALZ-Policies-Testing.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+#### 3. Write the Pester tests
+
+Write the Pester tests in the test file. The tests should cover the following scenarios:
+
+- Conditions that should be true when the policy is evaluated, so it is compliant
+- Conditions that should be false when the policy is evaluated, so it is non-compliant
+
+It is important to test all the conditions evaluated in the policy. More on this below.
+
+### Where is the testing framework?
```
Comment (jtracey93) [inline docs/wiki/ALZ-Policies-Testing.md:50]: Missing the workflow or an example one to call this.

Also the pre-reqs we have like the MG and policy defs deployed and subs moved.

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Looks good, just a few nit picks

File: docs/wiki/media/ALZ Policy Assignments v2.xlsx
[the remark this answers is not in the document]
Comment (Springstone) [reply docs/wiki/media/ALZ Policy Assignments v2.xlsx:1]: No it's right, I found two assignments with the incorrect effect documented, which I fixed.

File: docs/wiki/ALZ-Policies-Testing.md (hunk 1 above)
Comment (Springstone) [reply docs/wiki/ALZ-Policies-Testing.md:50]: Good shout, added some more text around this. @jtracey93

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**